### PR TITLE
irssi: fix syntax error when no channels are specified

### PR DESCRIPTION
**Pull request by ReplayCoding**
### Description

TL;DR: Don't nest concatStringsSep

`concatStringsSep` will return an empty string when called with empty array. The `concatStringsSep` called with `v.channels` would generate an empty string for each channel:

```
programs.irssi = {
  networks = {
    network1 = {
      ...
    };
    network2 = {
      ...
    };
  };
};
```
generates `[ "" "" ]`, which the `concatStringsSep` called with `v.networks` will use to generate
```
,

```

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

### CODE REVIEW

**Comment by rycee** [comment]
Thanks! Merged to master now.
